### DISCUSSION TITLE
`pageContext.urlOriginal` contains unusual `.pageContext.json`

### DISCUSSION THREAD

**Discussion by openscript** [vikejs/vike#1626] [Discussion] [answered]
Hey Vikeings

I'm not sure if I'm doing something wrong or not, but I'm struggling with `pageContext.urlOriginal`:

1. I navigate to `/en/` and `pageContext.urlOriginal` points to `/en/` in my global `+onBeforeRoute` as expected.
2. I use a link (HTML `<a>` inside the page) that leads to `/en/` again.
3. Inside the global `+onBeforeRoute` the `pageContext.urlOriginal` is set to `/en/index.pageContext.json`.

My settings are `{ prerender: true, disableUrlNormalization: true }` but it's the same when I deactivate them.

Any idea why this might is happening?

Thank you

**Comment by openscript** [comment]
It seems like it's linked to https://vike.dev/pageContext.json. What options I do have here?

1. Make everything client side?
2. Use some regular expression to remove `index.pageContext.json` from the url?
3. Use `pageContext.urlParsed` and puzzle together the URL is it is displayed in the address bar? This is tricky to do right, as `urlParsed` doesn't respect `disableUrlNormalization` regarding trailing slashes.

**Reply by brillout** [reply]
I'm not sure I understand your issue. What is it you want to achieve?

**Reply by openscript** [reply]
I've used `urlOriginal` and it contained `index.pageContext.json`, but I understand know, that I should use `urlParsed`.

**Reply by brillout** [reply, marked as answer]
FYI `pageContext.urlParsed` is going to be renamed `pageContext.url` hopefully further incentivizing users to use this instead of `urlOriginal`.
